[ENH] BEP 018 - Genetic Information (#395)
bids-standard/bids-specification: src/04-modality-specific-files/08-genetic-descriptor.md

# Genetic Descriptor

Support for genetic descriptors was developed as a [BIDS Extension
Proposal](../06-extensions.md#bids-extension-proposals).

> The extension was primarily developped by Cyril Pernet and Clara Moreau with

Reviewer: ```suggestion
The extension was primarily developed by Cyril Pernet and Clara Moreau with
```

| Field name           | Definition                                                                     |
|----------------------|--------------------------------------------------------------------------------|
| Genetics.Dataset     | REQUIRED. URI where data can be retrieved.                                     |
| Genetics.Database    | OPTIONAL. URI of database where the dataset is hosted.                         |
| Genetics.Descriptors | OPTIONAL. List of relevant descriptors (*e.g.*, journal articles) for dataset. |

Example:

```JSON
{
  "Name": "Human Connectome Project",
  "BIDSVersion":  "1.3.0",
  "License": "CC0",
  "Authors": ["1st author", "2nd author"],
  "Funding": ["P41 EB015894/EB/NIBIB NIH HHS/United States"],
  "Genetics": {
     "Dataset": "https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs001364.v1.p1",
     "Database": "https://www.ncbi.nlm.nih.gov/gap/",
     "Descriptors": ["https://doi.org/10.1016/j.neuroimage.2013.05.041"]
     }
}
```

## Subject naming and Participants file

If the same participants have different identifiers in the genetic and imaging datasets,

> the column `GeneticID` SHOULD be added to the `participants.tsv` file to associate

Author: @CPernet Is it very important to use `GeneticID` instead of `genetic_id`? It is inconsistent with TSV column naming conventions elsewhere in the spec.

Reviewer 2: we can use whatever we want, but this is used twice in the spec and it appears in the example, we just need to be consistent, now I'm not bothered if you prefer `GeneticID` or `genetic_id` pick what you think works best / is consistent with other specs naming

Reviewer 2: genetic_id everywhere pushed

the BIDS participant with a subject in the `Genetics.Dataset` referred to in the
`dataset_description.json` file.

Information about the presence/absence of specific genetic markers MAY be duplicated

> in the `participants.tsv` file by adding optional columns.

Reviewer 3: ```suggestion
in the `participants.tsv` file by adding optional columns (like `idh_mutation` in the
example below).
Note that optional columns MUST be further described in an accompanying
`participants.json` file as described in the [tabular files section](https://bids-specification.rtfd.io/en/stable/02-common-principles.html#tabular-files).
```

sub-control01	34	M	control	124587	yes
sub-control02	12	F	control	548936	yes
sub-patient01	33	F	patient	489634	no
```

## Genetic Information

Template:

```Text
genetic_info.json
```

The `genetic_info.json` file describes the genetic information available in the
`participants.tsv` file and/or the genetic database described in
`dataset_description.json`.
Datasets containing the `Genetics` field in `dataset_description.json` or the
`genetic_id` column in `participants.tsv` MUST include this file with the following
fields:

| Field name         | Definition                                                                  | Values                                                                                                                                                                                                             |
| :----------------- | :-------------------------------------------------------------------------- | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | GeneticLevel       | MANDATORY Describes the level of analysis                                   | `Genetic`, `Genomic`, `Epigenomic`, `Transcriptomic`, `Metabolomic`, or `Proteomic`                                                                                                                                |

Reviewer: ```suggestion
| GeneticLevel       | REQUIRED. Describes the level of analysis                                   | `Genetic`, `Genomic`, `Epigenomic`, `Transcriptomic`, `Metabolomic`, or `Proteomic`                                                                                                                                |
```

Consistent language with the rest of BIDS terminology

> | AnalyticalApproach | OPTIONAL Methodology or methodologies used to analyse the GeneticLevel      | String or list of strings. Each Value must be taken from [gapsolr][] under /Study/Molecular Data Type, for instance `SNP Genotypes (Array)` or `Methylation (CpG)`           |

Reviewer 4: Is [gapsolr][] supposed to link to another resource ?

Reviewer 4: As is, just the shortened link is appearing as the name and it seems a bit confusing ! But maybe that was intentional ?

Reviewer 2: nope it was an oversight that you fixed :-) thx

> | SampleOrigin       | MANDATORY Describes from which tissue the genetic information was extracted | `blood`, `saliva`, `brain`, `csf`, `breast milk`, `bile`, `amniotic fluid`, `other biospecimen`                                                                                                                    |

Reviewer: ```suggestion
| SampleOrigin       | REQUIRED. Describes from which tissue the genetic information was extracted | `blood`, `saliva`, `brain`, `csf`, `breast milk`, `bile`, `amniotic fluid`, `other biospecimen`                                                                                                                    |
```

perhaps moving this one line higher to group the required fields

| Genetics.Descriptors | OPTIONAL. List of relevant descriptors (*e.g.*, journal articles) for dataset. |

Example:

```JSON
{
  "Name": "Human Connectome Project",
  "BIDSVersion":  "1.3.0",
  "License": "CC0",
  "Authors": ["1st author", "2nd author"],
  "Funding": ["P41 EB015894/EB/NIBIB NIH HHS/United States"],
  "Genetics": {
     "Dataset": "https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs001364.v1.p1",
     "Database": "https://www.ncbi.nlm.nih.gov/gap/",
     "Descriptors": ["https://doi.org/10.1016/j.neuroimage.2013.05.041"]
     }
}
```

## Subject naming and Participants file

If the same participants have different identifiers in the genetic and imaging datasets,
the column `genetic_id` SHOULD be added to the `participants.tsv` file to associate
the BIDS participant with a subject in the `Genetics.Dataset` referred to in the
`dataset_description.json` file.

Information about the presence/absence of specific genetic markers MAY be duplicated
in the `participants.tsv` file by adding optional columns (like `idh_mutation` in the
example below).
Note that optional columns MUST be further described in an accompanying

> `participants.json` file as described in the [tabular files section](https://bids-specification.rtfd.io/en/stable/02-common-principles.html#tabular-files).

Author: ```suggestion
`participants.json` file as described in the [tabular files section](02-common-principles.md#tabular-files).
```

example below).
Note that optional columns MUST be further described in an accompanying
`participants.json` file as described in
[Tabular files](../02-common-principles.md#tabular-files).

`participants.tsv` example:

```Text
participant_id	age	sex	group	genetic_id	idh_mutation
sub-control01	34	M	control	124587	yes
sub-control02	12	F	control	548936	yes
sub-patient01	33	F	patient	489634	no
```

## Genetic Information

Template:

```Text
genetic_info.json
```

The `genetic_info.json` file describes the genetic information available in the
`participants.tsv` file and/or the genetic database described in
`dataset_description.json`.
Datasets containing the `Genetics` field in `dataset_description.json` or the
`genetic_id` column in `participants.tsv` MUST include this file with the following
fields:

| Field name         | Definition                                                                  | Values                                                                                                                                                                                                             |

> | :----------------- | :-------------------------------------------------------------------------- | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

Author: I seem to remember this colon (`:`) convention being removed from the rest of the document during PDFification. Is that the cause of the weird table formatting [here](https://2079-150465237-gh.circle-artifacts.com/0/bids-spec.pdf#43)?

Reviewer 3: yes, very likely. @Arshitha may have background info. I think there's no harm in removing the colons here as well. The HTML readability suffers only a little, while the PDF output is repaired.

Author: ```suggestion
|--------------------|-----------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
```

Reviewer 5: yes, the colons were removed from the pipe tables for generating the pdf. Also, it might be a  good idea to extend the column widths because when text in the columns are long, it could lead to overlapping between columns (in the pdf). If it'd be valuable, I could document the conventions  that I followed for pipe tables with pdf generation in mind? @sappelhoff

Reviewer 3: indeed that'd be helpful. Perhaps as an addition to the docs you wrote in https://github.com/bids-standard/bids-specification/blob/master/pdf_build_src/README.md ?

Reviewer 5: I looked at the pdf and there's indeed overlapping here. I don't have the permissions to the edit file. But increasing/extending the first column width by maybe 10 hyphens or so should fix it.

| SampleOrigin       | REQUIRED. Describes from which tissue the genetic information was extracted | `blood`, `saliva`, `brain`, `csf`, `breast milk`, `bile`, `amniotic fluid`, `other biospecimen`                                                                                                                    |
| TissueOrigin       | OPTIONAL Describes the type of tissue analyzed for SampleOrigin `brain`     | `gray matter`, `white matter`, `csf`, `meninges`, `macrovascular` or `microvascular`                                                                                                                               |
| BrainLocation      | OPTIONAL Refers to the location in space of the TissueOrigin                | `MNI coordinate` or a `label` taken from the [Allen Brain Atlas][allen] possibly `layer` to refer to layer-specific gene expression, which can also tie up with laminar fMRI                                       |

> | CellType           | OPTIONAL Describes the type of cell analyzed                                | Value should come from the [cell ontology][ontology]                                                                                                                                                               |

Author: Adding periods (`.`) after OPTIONAL/REQUIRED. Spaces removed to retain table alignment.

```suggestion
| AnalyticalApproach | OPTIONAL. Methodology or methodologies used to analyse the GeneticLevel     | String or list of strings. Each Value must be taken from the [database of Genotypes and Phenotypes (dbGaP)][gapsolr] under /Study/Molecular Data Type, for instance `SNP Genotypes (Array)` or `Methylation (CpG)` |
| SampleOrigin       | REQUIRED. Describes from which tissue the genetic information was extracted | `blood`, `saliva`, `brain`, `csf`, `breast milk`, `bile`, `amniotic fluid`, `other biospecimen`                                                                                                                    |
| TissueOrigin       | OPTIONAL. Describes the type of tissue analyzed for SampleOrigin `brain`    | `gray matter`, `white matter`, `csf`, `meninges`, `macrovascular` or `microvascular`                                                                                                                               |
| BrainLocation      | OPTIONAL. Refers to the location in space of the TissueOrigin               | `MNI coordinate` or a `label` taken from the [Allen Brain Atlas][allen] possibly `layer` to refer to layer-specific gene expression, which can also tie up with laminar fMRI                                       |
| CellType           | OPTIONAL. Describes the type of cell analyzed                               | Value should come from the [cell ontology][ontology]                                                                                                                                                               |
```

  
To ensure dataset description consistency, we recommend following [Multi-omics approaches to disease](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1215-1) by Hasin et al. 2017 to determine the `GeneticLevel:`   
- `Genetic`: data report on a single genetic location (typically directly in the `participants.tsv` file) 
- `Genomic`:  data link to participants' genome (multiple genetic locations)
- `Epigenomic`: data link to participants' characterization of reversible modifications of DNA
- `Transcriptomic`: data link to participants RNA levels
- `Metabolomic`: data link to participants' products of cellular metabolic functions

> - `Proteomic`: data link to participants peptides and proteins quantification 

Author: Appease the linter.

```suggestion
  - `Genetic`: data report on a single genetic location (typically directly in the `participants.tsv` file) 
  - `Genomic`:  data link to participants' genome (multiple genetic locations)
  - `Epigenomic`: data link to participants' characterization of reversible modifications of DNA
  - `Transcriptomic`: data link to participants RNA levels
  - `Metabolomic`: data link to participants' products of cellular metabolic functions
  - `Proteomic`: data link to participants peptides and proteins quantification 
```
